Templates/templates/PHP/Model/EntityType.php.tt

```diff
@@ -94,15 +94,22 @@ foreach(var property in entity.Properties.Where(prop => prop.Type.GetTypeString(
     * <#=property.GetSanitizedLongDescription()#>
 <# } #>
      *
-     * @return array|null The <#=propertyName#>
+     * @return <#=propertyTypeReference#>[]|null The <#=propertyName#>
      */
     public function get<#=propertyName.ToCheckedCase()#>()
     {
-        if (array_key_exists("<#=camelCasePropertyName#>", $this->_propDict)) {
-           return $this->_propDict["<#=camelCasePropertyName#>"];
-        } else {
-            return null;
+        if (array_key_exists('<#=camelCasePropertyName#>', $this->_propDict) && !is_null($this->_propDict['<#=camelCasePropertyName#>'])) {
+           $<#=camelCasePropertyName#> = [];
+           if (count($this->_propDict['<#=camelCasePropertyName#>']) > 0 && is_a($this->_propDict['<#=camelCasePropertyName#>'][0], '<#=propertyTypeReference#>')) {
```

Comment:
Why is this logic different than in ComplexType.php.tt? I think it should be the same.

Reply:
Updated the logic to match in the two files.